bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,380 @@
+        }
+
+        if (host != null && port > 0) {
+            // Create a socket to eGate
+            synchronized (this.lock) {
+                try {
+                    egateSocket = new Socket();
+                    egateSocket.connect(new InetSocketAddress(host, port));
+                    egateSocket.setSoTimeout(SOCKET_TIMEOUT);
+                    writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+                    writer.write("SilenceModeSet;Value=0;" + CR);
+                    writer.flush();
+
+                } catch (UnknownHostException e) {
+                    logger.error("unknown socket host {}", host);
+                    try {
+                        egateSocket.close();
+                        updateStatus(ThingStatus.OFFLINE);
+                    } catch (IOException e1) {
+                        if (logger.isErrorEnabled()) {
+                            logger.error("EGate Socket not closed {}", e1.toString());
+                        }
+                    }
+                    egateSocket = null;
+                } catch (SocketException e) {
+                    if (logger.isErrorEnabled()) {
+                        logger.error("{}", e.getLocalizedMessage());
+                    }
+                    try {
+                        egateSocket.close();
+                        updateStatus(ThingStatus.OFFLINE);
+                    } catch (IOException e1) {
+                        if (logger.isErrorEnabled()) {
+                            logger.error("EGate Socket not closed {}", e1.toString());
+                        }
+                    }
+                    egateSocket = null;
+                } catch (Exception e) {
+                    if (logger.isErrorEnabled()) {
+                        logger.error(
+                                "Error while establishing connection to Dominoswiss eGate Server with Port: {} and IP: {} Error: {}",
+                                port, host, e.toString());
+                    }
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+                if (egateSocket != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                startAutomaticRefresh();
+                if (logger.isDebugEnabled()) {
+                    logger.debug("EGate Handler connected and online, Status {} ",
+                            this.getThing().getStatus().toString());
+                }
+            }
+        } else {
+            if (logger.isErrorEnabled()) {
+                logger.error("Invalid IP address for dominoswiss eGate or wrong port : '{}'/'{}'", host, port);
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to dominoswiss eGate gateway. host IP address or port are not set.");
```

Comment:
Dont need to use logger here since you are changing the thingstatus which will cause an event.log to occur already. See here
https://www.openhab.org/docs/developer/guidelines.html#f-logging